api2-war/src/main/java/eu/europeana/api2/v2/web/controller/ContentReuseFrameworkController.java

```diff
@@ -42,7 +71,99 @@
+                                             @RequestParam(value = "url", required = true) String url,
+                                             @RequestParam(value = "wskey", required = true) String wskey,
+                                             @RequestParam(value = "callback", required = false) String callback,
+                                             HttpServletRequest request,
+                                             HttpServletResponse response) {
+        long t0 = System.currentTimeMillis();
+        controllerUtils.addResponseHeaders(response);
+        LimitResponse limitResponse = null;
+        try {
+            limitResponse = controllerUtils.checkLimit(wskey, request.getRequestURL().toString(),
+                                                       "record.json", RecordType.OBJECT, null);
+        } catch (ApiLimitException e) {
+            response.setStatus(e.getHttpStatus());
+            return JsonUtils.toJson(new ApiError(e), callback);
+        }
+
+        CrfMetadataResult result = new CrfMetadataResult(wskey, "metadata-by-url.json", limitResponse.getRequestNumber());
+        SourceDocumentReferenceMetaInfo info = crfService.getMetadata(url);
+        if (info != null) {
+            result.imageMetaInfo = info.getImageMetaInfo();
+        }
+        result.statsDuration = (System.currentTimeMillis() - t0);
+        return JsonUtils.toJson(result, callback);
+    }
+
+    @RequestMapping(value = "/v2/thumbnail-by-url.json", method = RequestMethod.GET)
+    public ResponseEntity<byte[]> thumbnailByUrl(
+                                                        @RequestParam(value = "uri", required = true) String url,
+                                                        @RequestParam(value = "size", required = false, defaultValue = "FULL_DOC") String size,
+                                                        @RequestParam(value = "type", required = false, defaultValue = "IMAGE") String type,
+                                                        @RequestParam(value = "callback", required = false) String callback,
+                                                        HttpServletRequest request,
+                                                        HttpServletResponse response) {
+        controllerUtils.addResponseHeaders(response);
+
+        log.info("Thumbnail");
+
+        String sufix = "";
+        log.info("Size: " + size);
+        log.info("Type: " + type);
+
+        if(size.equalsIgnoreCase("BRIEF_DOC") || size.equalsIgnoreCase("h180") ||
+           size.equalsIgnoreCase("FULL_DOC") || size.equalsIgnoreCase("FULL_DOC_ALL") || size.equalsIgnoreCase("w200")) {
+            sufix = "200";
+        }
+        else if (size.equalsIgnoreCase("w400")) {
+            sufix = "400";
+        }
+
+        final HttpHeaders headers = new HttpHeaders();
+
+        byte[] imageResponse = DefaultImage.getImage(ThumbSize.LARGE, DocType.IMAGE);
+        if(type.equalsIgnoreCase("IMAGE")) {
```

Comment:
A equalsIgnoreCase is nothing else then a X.toUppercase().equals(Y.toUppercase())... So by uppercasing in the switch (or lowercasing, whatever you prefer) and upper or lowercasing the possible values in the cases, you come to the same result and can perfectly use a string switch.

No, if you use proper layout formatting, I think a switch case is more readible then a bundle of if elses. In this case especially because inside some cases you have another if else structure.

In this code example I was talking about the type cases, not the size cases as those if statements are a bit more complex.